src/store_client.cc

```diff
@@ -899,7 +897,7 @@ store_client::dumpStats(MemBuf * output, int clientNumber) const
     if (_callback.pending())
         return;
 
-    output->appendf("\tClient #%d, %p\n", clientNumber, _callback.callback_data);
+    output->appendf("\tClient #%d, %p\n", clientNumber, this);
```

Comment:
This adjustment changes which raw pointer is printed for each store_client in `mgr:objects`, `mgr:vm_objects`, and `mgr:openfd_objects` reports. The printed pointer can still be used to identify the corresponding storeClientCopy() caller in a debugging log. In some sense, printing `this` is better because a single storeClientCopy() caller often owns multiple store_client objects at the same time, but that is not why I made this change. I made this otherwise unnecessary/out-of-scope change because CallbackData API does not expose a raw pointer value.

We could enhance CallbackData API, of course, but exposing that raw pointer is a bit dangerous, and we should refactor to give store_client a unique instance ID instead of using frequently recycled memory addresses anyway. That refactoring needs quite a bit of noisy out-of-scope code changes, including replacing all these ugly appendf() calls with std::ostream-based code.

Reply:
FTR, displaying the raw-pointer value as an ID here is a security issue. It exposes operating system internal layouts to external viewers, violating OS memory protections. If there is any other abstraction we can use as a unique client ID that would be much better.
I do not insist on an ID mechanism being created yet, but if an alternative is already available please switch to that value instead of `this`.